src/utils/fetchZaps.ts

```diff
@@ -316,3 +316,60 @@ export const fetchNostrProfile: ResourceFetcher<
         throw new Error("Failed to load profile");
     }
 };
+
+// Search results from primal have some of the stuff we want for a TagItem contact
+export type PseudoContact = {
+    name: string;
+    hexpub: string;
+    ln_address?: string;
+    image_url?: string;
+};
+
+export async function searchProfiles(query: string): Promise<PseudoContact[]> {
+    console.log("searching profiles...");
+    const response = await fetch(PRIMAL_API, {
+        method: "POST",
+        headers: {
+            "Content-Type": "application/json"
+        },
+        body: JSON.stringify([
+            "user_search",
+            { query: query.trim(), limit: 10 }
+        ])
+    });
+
+    if (!response.ok) {
+        throw new Error(`Failed to search`);
+    }
+
+    const data = await response.json();
+
+    const users: PseudoContact[] = [];
+
+    for (const object of data) {
+        if (object.kind === 0) {
+            try {
+                const content = JSON.parse(object.content);
+                const contact: Partial<PseudoContact> = {
+                    hexpub: object.pubkey
+                };
+                if (content?.name) {
+                    contact.name = content.name;
+                } else if (content?.display_name) {
+                    contact.name = content.display_name;
+                }
```

Comment:
in mutiny-node we do display_name then try name, should have the logic the same here

Reply:
yeah let's unify this somewhere but not now

Reply:
In damus my `display_name` is Carman 🏴‍☠ but my `name` is `@benthecarman` so I definitely think we should use `display_name` here

Reply:
find you convinced me